File: Java/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.Arrays;
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int zero_cnt = (int)Arrays.stream(lottos).filter(num -> num == 0).count();
        //Arrays.sort(lottos);
        Arrays.sort(lottos);
        int dang = 0;
        for(int i=0; i<win_nums.length;i++){
            for(int j=zero_cnt; j<lottos.length;j++){
                if(win_nums[i]==lottos[j]){
                    dang+=1;
                }
            }
        }
        int answer[] = new int[2];
        answer[0]= 7 - (dang+zero_cnt) > 5 ? 6 : 7 - (dang+zero_cnt);
        answer[1]= 7 - dang > 5 ? 6 : 7 - dang;
        return answer;
    }
}

File: Kotlin/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.kt
class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        val zeroCount = lottos.count { it == 0 }
        val matchCount = lottos.count { it != 0 && it in win_nums }
        
        return intArrayOf(rank(matchCount+zeroCount),rank(matchCount))
    }
    fun rank(ok : Int) : Int {
        return when {
            ok >= 6 -> 1
            ok == 5 -> 2
            ok == 4 -> 3
            ok == 3 -> 4
            ok == 2 -> 5
            else -> 6
        }
    }
}